packages/Bio/src/tests/detectors-tests.ts

```diff
@@ -195,72 +207,72 @@ MWRSWY-CKHP
     return df;
   }
 
-  const _csvDfs: { [key: string]: Promise<DG.DataFrame> } = {};
-  const readCsv: (key: string, csv: string) => DfReaderFunc = (key: string, csv: string) => {
+  const readCsv: (key: csvTests) => DfReaderFunc = (key: keyof typeof csvData) => {
     return async () => {
-      if (!(key in _csvDfs)) {
-        _csvDfs[key] = (async (): Promise<DG.DataFrame> => {
-          const df: DG.DataFrame = DG.DataFrame.fromCsv(csv);
-          await grok.data.detectSemanticTypes(df);
-          return df;
-        })();
-      }
-      return _csvDfs[key];
+      // Always recreate test data frame from CSV for reproducible detector behavior in tests.
+      const csv: string = csvData[key];
+      const df: DG.DataFrame = DG.DataFrame.fromCsv(csv);
+      await grok.data.detectSemanticTypes(df);
+      return df;
     };
   };
 
 
-  test('NegativeEmpty', async () => { await _testNeg(readCsv('csvDfEmpty', csvDfEmpty), 'col1'); });
-  test('Negative1', async () => { await _testNeg(readCsv('csvDf1', csvDf1), 'col1'); });
-  test('Negative2', async () => { await _testNeg(readCsv('csvDf2', csvDf2), 'col1'); });
-  test('Negative3', async () => { await _testNeg(readCsv('csvDf3', csvDf3), 'col1'); });
-  test('NegativeSmiles', async () => { await _testNeg(readCsv('csvDfSmiles', csvDfSmiles), 'col1'); });
+  test('NegativeEmpty', async () => { await _testNeg(readCsv(csvTests.negEmpty), 'col1'); });
```

Comment:
you use `'col1'` and `'seq'` multiple times, maybe it makes sense to put it into const?